src/component/MovieForm/MovieForm.js

```diff
@@ -1,60 +1,140 @@
 import React, { useState } from "react";
 import "./MovieForm.css";
 import {
-  COLUMN_CONFIG,
+  DURATION,
+  GENRE,
+  MOVIE_URL,
   OVERVIEW,
+  RATING,
+  RELEASE_YEAR,
   RESET,
-  SUBMIT
+  SUBMIT,
+  TITLE
 } from "../../constants";
+import { useForm } from 'react-hook-form';
 
 const MovieForm = ({movieInfo, onSubmit }) => {
- 
-  const [formData, setFormData] = useState(movieInfo);
+  const { register, handleSubmit, formState: { errors }, setValue } = useForm({
+    defaultValues: movieInfo,
+  });
 
-  const handleInputChange = (e) => {
-    setFormData({
-      ...formData,
-      [e.target.name]: e.target.value,
-    });
-  };
-
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    if (onSubmit) {
-      onSubmit(formData);
+  const handleFormSubmit = (data) => {
+    let genresList = data.genres;
+    if(!Array.isArray(genresList)) {
+      genresList = genresList.split(',');
     }
+
+    const newData = { ...data, genres: genresList, vote_average: parseInt(data.vote_average),runtime: parseFloat(data.runtime) };
+    onSubmit(newData);
   };
 
   const handleReset = () => {
-    setFormData(movieInfo);
+    if (movieInfo) {
+      Object.keys(movieInfo).forEach((key) => {
+        setValue(key, movieInfo[key]);
+      });
+    }
   };
 
+  const handleGenreInputChange = (event) => {
+    const genres = event.target.value.split(',');
+    setValue('genres', genres);
+  }
+
   return (
     <div className="movie-form_container">
-        <form onSubmit={handleSubmit} onReset={handleReset}>
+        <form onSubmit={handleSubmit(handleFormSubmit)} onReset={handleReset}>
```

Comment:
You could rename handleFormSubmit to onSubmit

Reply:
The onSubmit form is a prop coming in from parent components `AddMovie` and `UpdateMovie`. That is why I have used handleFormSubmit name.